TestScheduleTestSuite/TestDisable is unstable

## Flaky Test

### Which jobs are failing
```
2023-11-29T01:42:11.6662377Z --- FAIL: TestScheduleTestSuite (103.50s)
2023-11-29T01:42:11.6662663Z     --- FAIL: TestScheduleTestSuite/TestDisable (26.11s)
2023-11-29T01:42:11.6662936Z         testutil.go:265: start to run test in pd mode
2023-11-29T01:42:11.6663205Z         testutil.go:269: finish to run test in pd mode
2023-11-29T01:42:11.6663459Z         testutil.go:274: start to run test in api mode
2023-11-29T01:42:11.6663618Z         testutil.go:64: 
2023-11-29T01:42:11.6664122Z             	Error Trace:	/home/runner/work/pd/pd/pkg/utils/testutil/testutil.go:64
2023-11-29T01:42:11.6664822Z             	            				/home/runner/work/pd/pd/tests/server/api/scheduler_test.go:686
2023-11-29T01:42:11.6665527Z             	            				/home/runner/work/pd/pd/tests/server/api/scheduler_test.go:604
2023-11-29T01:42:11.6666067Z             	            				/home/runner/work/pd/pd/tests/testutil.go:279
2023-11-29T01:42:11.6666613Z             	            				/home/runner/work/pd/pd/tests/testutil.go:260
2023-11-29T01:42:11.6667373Z             	            				/home/runner/work/pd/pd/tests/server/api/scheduler_test.go:568
2023-11-29T01:42:11.6667633Z             	Error:      	Condition never satisfied
2023-11-29T01:42:11.6667935Z             	Test:       	TestScheduleTestSuite/TestDisable
2023-11-29T01:42:11.6668017Z FAIL
```

### CI link
https://github.com/tikv/pd/actions/runs/7026828089/job/19120231295?pr=7465

### Reason for failure (if possible)


### Anything else

--- a/pkg/schedule/coordinator.go
+++ b/pkg/schedule/coordinator.go
@@ -513,6 +513,7 @@ func (c *Coordinator) InitSchedulers(needRun bool) {
 	if err := c.cluster.GetSchedulerConfig().Persist(c.cluster.GetStorage()); err != nil {
 		log.Error("cannot persist schedule config", errs.ZapError(err))
 	}
+	log.Info("scheduler config is updated", zap.Reflect("scheduler-config", scheduleCfg.Schedulers))
 
 	c.markSchedulersInitialized()
 }

--- a/server/cluster/cluster.go
+++ b/server/cluster/cluster.go
@@ -467,6 +467,10 @@ func (c *RaftCluster) startGCTuner() {
 func (c *RaftCluster) runStoreConfigSync() {
 	defer logutil.LogPanic()
 	defer c.wg.Done()
+	// TODO: After we fix the atomic problem of config, we can remove this failpoint.
+	failpoint.Inject("skipStoreConfigSync", func() {
+		failpoint.Return()
+	})
 
 	var (
 		synced, switchRaftV2Config, needPersist bool
@@ -491,6 +495,7 @@ func (c *RaftCluster) runStoreConfigSync() {
 			if err := c.opt.Persist(c.storage); err != nil {
 				log.Warn("store config persisted failed", zap.Error(err))
 			}
+			log.Info("store config is updated")
 		}
 		select {
 		case <-c.ctx.Done():

--- a/tests/server/api/scheduler_test.go
+++ b/tests/server/api/scheduler_test.go
@@ -564,8 +564,10 @@ func (suite *scheduleTestSuite) checkAPI(cluster *tests.TestCluster) {
 }
 
 func (suite *scheduleTestSuite) TestDisable() {
+	suite.NoError(failpoint.Enable("github.com/tikv/pd/server/cluster/skipStoreConfigSync", `return(true)`))
 	env := tests.NewSchedulingTestEnvironment(suite.T())
 	env.RunTestInTwoModes(suite.checkDisable)
+	suite.NoError(failpoint.Disable("github.com/tikv/pd/server/cluster/skipStoreConfigSync"))
 }
 
 func (suite *scheduleTestSuite) checkDisable(cluster *tests.TestCluster) {

